### PR TITLE
Add eslint for linting

### PR DESCRIPTION
**Pull request by kitsonk**
This PR adds ESLint and associated configuration for CI as well as an initial tranche of linting changes.

### CODE REVIEW

File: log/logger.ts [hunk 1]
```diff
@@ -22,7 +24,8 @@ export class Logger {
     this.handlers = handlers || [];
   }
 
-  _log(level: number, msg: string, ...args: any[]) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _log(level: number, msg: string, ...args: any[]): void {
```
Comment (j-f1) [inline log/logger.ts]: Could this be `unknown[]` instead so type safety isn’t lost?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,31 +1,37 @@
 variables:
-  DENO_VERSION: 'v0.3.1'
+  DENO_VERSION: "v0.3.1"
+  TS_VERSION: "3.2.1"
 
 jobs:
```
Comment (ry) [inline azure-pipelines.yml]: Add a TODO to DRY up this file

File: log/logger.ts (hunk 1 above)
Comment (kitsonk) [reply log/logger.ts]: Possibly, something we can try to clean up later.  There is quite a bit of stuff that doesn't work under `strict` mode either... Ry and I agreed to stop trying to land all the cleanup in this and just get it running.

File: azure-pipelines.yml (hunk 2 above)
Comment (kitsonk) [reply azure-pipelines.yml]: Done

File: benching/mod.ts [hunk 3]
```diff
@@ -14,10 +14,10 @@ export interface BenchmarkTimer {
 }
 
 /** Defines a benchmark through a named function. */
-export type BenchmarkFunction = {
+export interface BenchmarkFunction {
```
Comment (ry) [inline benching/mod.ts]: oof wtf...

Comment (kitsonk) [reply benching/mod.ts]: The defaults for eslint for TypeScript specify that when something looks like an interface, it is preferred over a type alias.  I agree with that default.

Comment (ry) [reply benching/mod.ts]: Oh, I didn't even know about the other way of specifying an interface. The "wtf" was meant more to myself like: "wtf I approved this?"

**Pull request merged**